go/src/meguca/db/admin.go

```diff
@@ -263,3 +274,30 @@ func GetModLog(board string) (log []auth.ModLogEntry, err error) {
 	)
 	return
 }
+
+// GetPostModLog retrieves a post's (3) relevant mod-log entries
+func GetPostModLog(id uint64) []auth.ModLogEntry {
+	log := make([]auth.ModLogEntry, 3, 3)
+
+	for i, val := range [3]auth.ModerationAction {
+		auth.BanPost,
+		auth.DeletePost,
+		auth.MeidoVision,
+	} {
+		sq.Select("type", "id", "length", "created", "board", "by", "reason").
```

Comment:
The lack of error checking is intentional, since it's not guaranteed for a mod-log entry to still exist.
That's why in calls to this function, log[whatever].ID is checked if not 0.

Reply:
>That's why in calls to this function, log[whatever].ID is checked if not 0.

That's why  __the return value__ is checked if not 0 in log[whatever].ID.*

Reply:
So if it's sql no rows. Okay.